### ISSUE TITLE
TypeError: invalid arguments

### ISSUE THREAD

**Issue by JohnHalfords** [caronc/nzbget-subliminal#11]
Hello,

I get the error "TypeError: invalid arguments".
Hope you can help me.

Log:
INFO	2015-04-18 17:18:02	Executing post-process-script Subliminal.py for Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get_files('/share/Download/nzb/Movies/_Test') with 1 filter(s)
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Allowed Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.nzb.queued (regex)
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(system) NZBDIR="/share/Download/nzb"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get_files('/share/Download/nzb') with 1 filter(s)
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] NZB-Files located: Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.nzb.queued (2015-04-18 17:11:18)
INFO	2015-04-18 17:18:07	Subliminal: NZB-File detected: Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.nzb.queued
INFO	2015-04-18 17:18:07	Subliminal: NZBParse - NZB-File parsed 4 meta entries
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(system) NZBID="153"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) VIDEOEXTENSIONS=".mkv,.avi,.divx,.xvid,.mov,.wmv,.mp4,.mpg,.mpeg,.vob,.iso"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) MINSIZE="150"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) OVERWRITE="no"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) SINGLE="yes"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) UPDATETIMESTAMP="yes"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) UPDATEPERMISSIONS="no"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) VIDEOPERMISSIONS="644"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get_files('/share/Download/dst/Movies/_Test/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG') with 11 filter(s)
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Allowed Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.avi (suffix)
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Allowed A4A.mkv (suffix)
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Filtered "/share/Download/dst/Movies/_Test/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG/Waar vind je de Angels/A4A.mkv" (21.74 MB)
INFO	2015-04-18 17:18:07	Subliminal: Found 1 matched file(s).
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Scanning "/share/Download/dst/Movies/_Test/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG/MOVIE@720p-Xvid/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.avi" (2129.23 MB)
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Video timestamp updated.
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(system) TEMPDIR="/share/Download/tmp"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) CACHEDIR="/share/Download/tmp/subliminal"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) FORCEENCODING="None"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) MINSCORE="20"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) SKIPEMBEDDED="yes"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) SEARCHMODE="advanced"
INFO	2015-04-18 17:18:07	Subliminal: Using advanced search mode
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) MOVIEPROVIDERS=""
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) PROVIDERS="opensubtitles, tvsubtitles, podnapisi, addic7ed, thesubdb"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) TVSHOWPROVIDERS=""
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) PROVIDERS="opensubtitles, tvsubtitles, podnapisi, addic7ed, thesubdb"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Using the following movie providers: podnapisi, opensubtitles, addic7ed, thesubdb, tvsubtitles
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Using the following tv show providers: podnapisi, opensubtitles, addic7ed, thesubdb, tvsubtitles
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) LANGUAGES="nl"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) FETCHMODE="StandardFirst"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Fetch Mode: StandardFirst
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get_files('/share/Download/dst/Movies/_Test/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG/MOVIE@720p-Xvid') with 1 filter(s)
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Scanning [advanced] using /share/Download/dst/Movies/_Test/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG/MOVIE@720p-Xvid/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.avi lang=nl
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] get(config) TVCATEGORIES="tv; tv2; Series"
INFO	2015-04-18 17:18:07	Subliminal: [DEBUG] Deobfuscate - Stripped filename down to: MOVIE@720p-Xvid/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.avi
INFO	2015-04-18 17:18:08	Subliminal: [DEBUG] get(config) SYSTEMENCODING="UTF-8"
INFO	2015-04-18 17:18:08	Subliminal: [DEBUG] Guessing using: MOVIE@720p-Xvid/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.avi
ERROR	2015-04-18 17:18:13	Subliminal: Traceback (most recent call last):
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal/nzbget/ScriptBase.py", line 2295, in run
INFO	2015-04-18 17:18:13	Subliminal:     exit_code = main_function(*args, **kwargs)
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal.py", line 1330, in postprocess_main
INFO	2015-04-18 17:18:13	Subliminal:     overwrite=overwrite,
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal.py", line 1026, in subliminal_fetch
INFO	2015-04-18 17:18:13	Subliminal:     use_nzbheaders=use_nzbheaders,
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal.py", line 555, in guess_info
INFO	2015-04-18 17:18:13	Subliminal:     opts={'nolanguage': True, 'nocountry': True},
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal/guessit/matcher.py", line 110, in __init__
INFO	2015-04-18 17:18:13	Subliminal:     from guessit.plugins import transformers
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal/guessit/plugins/transformers.py", line 212, in <module>
INFO	2015-04-18 17:18:13	Subliminal:     reload()
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal/guessit/plugins/transformers.py", line 210, in reload
INFO	2015-04-18 17:18:13	Subliminal:     reload_options(all_transformers())
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal/guessit/options.py", line 70, in reload
INFO	2015-04-18 17:18:13	Subliminal:     _opts, _naming_opts, _output_opts, _information_opts, _webservice_opts, _other_opts = build_opts(transformers)
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal/guessit/options.py", line 63, in build_opts
INFO	2015-04-18 17:18:13	Subliminal:     transformer.register_options(opts, naming_opts, output_opts, information_opts, webservice_opts, other_opts)
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/Download/ppscripts/Subliminal/guessit/transfo/expected_series.py", line 37, in register_options
INFO	2015-04-18 17:18:13	Subliminal:     help='List of expected series to parse. Separate series names with ";"')
ERROR	2015-04-18 17:18:13	Subliminal:   File "/share/HDA_DATA/.qpkg/Python/lib/python2.7/optparse.py", line 1018, in add_option
INFO	2015-04-18 17:18:13	Subliminal:     raise TypeError, "invalid arguments"
ERROR	2015-04-18 17:18:13	Subliminal: TypeError: invalid arguments
ERROR	2015-04-18 17:18:13	Post-process-script Subliminal.py for Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG failed
INFO	2015-04-18 17:18:13	Executing post-process-script Logger.py for Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG

**Comment by caronc** [comment]
Hmm you stumped me on this one; that same file works for me (here is how i tested it):
```bash
$ mkdir -p "tmp/_Test/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG/MOVIE@720p-Xvid/"
$ touch tmp/_Test/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG/MOVIE\@720p-Xvid/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.avi
$ ./Subliminal.py -f -s -c 10 -z 0 -p podnapisi -S tmp/_Test/
```
The output looked as follows:
```
2015-04-18 15:46:17,889 - 2235 - INFO - Found 1 matched file(s).
2015-04-18 15:46:17,889 - 2235 - INFO - Using advanced search mode
2015-04-18 15:46:18,146 - 2235 - INFO - Scanning video 'Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.avi' in 'tmp/_Test/Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG'
2015-04-18 15:46:18,146 - 2235 - WARNING - Size is lower than 10MB: hashes not computed
2015-04-18 15:46:19,008 - 2235 - INFO - Listing subtitles with provider 'podnapisi' for video <Movie [u'Two Night Stand', 2014]> with languages set([<Language [en]>])
2015-04-18 15:46:19,857 - 2235 - WARNING - Podnapisi page matches: 1
2015-04-18 15:46:19,873 - 2235 - INFO - Found 5 subtitle(s) on podnapisi
2015-04-18 15:46:20,646 - 2235 - INFO - Downloading subtitle <PodnapisiSubtitle [en]> with score 20 into u'Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.srt'
2015-04-18 15:46:22,781 - 2235 - INFO - Successfully placed Two.Night.Stand.2014.LIMITED.720p.BRRip.XviD.AC3-RARBG.srt
```
Could you try these commands as well for me and let me know what the output looks like?

Also:
* Is this happening to you all the time, or does it just happen on some files?
* What operating system are you using (with the Subliminal plugin).
* I can see that your using Python v2.7; do you know the exact version? ```python --version```

**Comment by JohnHalfords** [comment]
Hi, thansk for quick answer !
I did what you've asked:
- I have a Qnap Nas with NZBGET 12.0 and Python 2.7
- The problem is with all the files
- I'm willing to install NZBGET 13 or 14 but i'm now working with a version Qnap provides within QPKG.
- The output for the commands you gave:
[/share/HDA_DATA/Download/dst] # ../ppscripts/Subliminal.py -f -s -c 10 -z 0 -p podnapisi -S tmp/_Test/
2015-04-19 15:53:20,051 - 8144 - INFO - Found 1 matched file(s).
2015-04-19 15:53:20,054 - 8144 - INFO - Using advanced search mode
2015-04-19 15:53:23,287 - 8144 - ERROR - Fatal Exception:
  Traceback (most recent call last):
    File "../ppscripts/Subliminal/nzbget/ScriptBase.py", line 2295, in run
    exit_code = main_function(*args, **kwargs)
    File "../ppscripts/Subliminal.py", line 1471, in main
    use_nzbheaders=False,
    File "../ppscripts/Subliminal.py", line 1026, in subliminal_fetch
    use_nzbheaders=use_nzbheaders,
    File "../ppscripts/Subliminal.py", line 555, in guess_info
    opts={'nolanguage': True, 'nocountry': True},
    File "../ppscripts/Subliminal/guessit/matcher.py", line 110, in __init__
    from guessit.plugins import transformers
    File "../ppscripts/Subliminal/guessit/plugins/transformers.py", line 212, in <module>
    reload()
    File "../ppscripts/Subliminal/guessit/plugins/transformers.py", line 210, in reload
    reload_options(all_transformers())
    File "../ppscripts/Subliminal/guessit/options.py", line 70, in reload
    _opts, _naming_opts, _output_opts, _information_opts, _webservice_opts, _other_opts = build_opts(transformers)
    File "../ppscripts/Subliminal/guessit/options.py", line 63, in build_opts
    transformer.register_options(opts, naming_opts, output_opts, information_opts, webservice_opts, other_opts)
    File "../ppscripts/Subliminal/guessit/transfo/expected_series.py", line 37, in register_options
    help='List of expected series to parse. Separate series names with ";"')
    File "/share/HDA_DATA/.qpkg/Python/lib/python2.7/optparse.py", line 1018, in add_option
    raise TypeError, "invalid arguments"
  TypeError: invalid arguments

Have a nice sunday,
Grtx, John Halfords

**Comment by caronc** [comment]
Interesting enough, [you're not the same person with this problem](http://nzbget.net/forum/viewtopic.php?f=3&t=1880&start=10).  It appears the [guessit](https://github.com/wackou/guessit) library (that Subliminal also uses) doesn't like certain version python v2.7 and was apparently resolved in this thread [here](https://github.com/wackou/guessit/issues/178). More specifically, [here is exactly your issue](https://github.com/wackou/guessit/issues/171).

I'll see if it's something I can easily back port, otherwise I'll see if it's a simple task to update to guessit v0.10.0. Hopefully i can look into this next weekend; if i can, i'll look into it earlier.

**Comment by JohnHalfords** [comment]
Thanks Chris,

Anyway, i'm workin' on my Rasberry Pi to be my download server now. Then i
can use NZBGet 14.2...

Regards, John Halfords

2015-04-21 0:59 GMT+02:00 Chris <notifications@github.com>:

**Comment by caronc** [comment]
John,

Let me know how things go,  I bundled up a new tag ([v0.9.5](https://github.com/caronc/nzbget-subliminal/releases/tag/v0.9.5)) which includes this fix.  You _should_ be good to go.  I'll close the ticket when you're satisfied! :)

**Comment by JohnHalfords** [comment]
Hello Chris,

As i told you, i installed NZBGet 14.2 on my raspPi, thats working very
good, indeed (as you told me) 14.2 has a lot of improvements comparing with
12.0. Your script is working fantastic under 14.2. I'm very happy with it.

The 12.0 is stil running on my Qnap Nas (under QPKG) so when it helps you i
am happy to test your 095 on the NZBGet 12.0.

I'm in Ostenreich now on holiday and next weekend i'm back in the
Netherlands, I will test it then !

For now; good to see you take your service very serious, thanks for that !

Regards, John

Regards, John Halfords

2015-04-28 1:11 GMT+02:00 Chris <notifications@github.com>:

**Comment by JohnHalfords** [comment]
Hello Chris,

I'm not able to test it anymore.
My NAS is hanging on something else...

I'm sorry.

Regards, John Halfords

2015-04-28 18:03 GMT+02:00 John Halfords <hajehajo@zonnet.nl>:

**Comment by caronc** [comment]
@JohnHalfords; don't worry about it at all. You're original problem is clearly identified an issue with guessit (a library my wrapper greatly relies on). I appreciate you reporting it and guessit developers have marked the issue as resolved with their latest version. Since I've updated to that, I'm comfortable in presuming this bug has been resolved.

If you ever do get around to testing it and it fails, I'll reopen this ticket.  Just let me know! That all said, I'm glad your Raspberry Pi has given you a working alternative! :)
